slapo/sharding/infer_type.py

```diff
@@ -133,3 +152,34 @@ def _batchnorm2d(module, sharded_size, axis):
         raise ValueError("BatchNorm2d only supports sharding on axis 0")
     module.num_features = sharded_size
     return ("partition", 1)
+
+
+@register_output_infer_fn(Conv1D)
+def _conv1d(module, param_name, sharded_size, axis):
+    """Output inference for Conv1D layer. Note that Conv1D has a transposed
+    weight (input features, output features) compared to Linear.
+    It adjusts the input or output feature size to reflect the shard size,
+    and returns the output type (partial or partition) after sharding.
+
+    Parameters
+    ----------
+    module: torch.nn.Linear
+        The linear layer to shard.
+    param_name: str
+        The name of the parameter to shard.
+    sharded_size: int
+        The size of the sharded dimension.
+    axis: int
+        The axis to shard on.
+
+    Returns
+    -------
+    tuple
+        The output type and the axis (if the output is partitioned).
+    """
+    if axis == 1 or param_name == "bias":
```

Comment:
why a special handle for bias? i don't see this in linear

Reply:
This is a bit tricky. The layout in Linear's weight is `[output feature, input feature]`, so sharding output for Linear would be:
```
.sync("weight", axis=0)
.sync("bias", axis=0)
```
So we previously don't check the parameter name because axis=0 always points to output feature.

However, the weight layout of Conv1D is transposed `[input feature, output feature]`, so sharding becomes:
```
.sync("weight", axis=1)
.sync("bias", axis=0)
```

In this case, we have to make sure sharding bias always means to shard the output.